Loader/ComboLoader.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * File containing the ComboLoader class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Loader;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+
+/**
+ * ComboLoader combines JS or CSS files.
+ */
+class ComboLoader implements Loader
+{
+    /**
+     * @var string Path to YUI build directory.
+     */
+    private $yuiBuildDirectory;
+
+    /**
+     * @var string Path to web directory.
+     */
+    private $webDirectory;
+
+    /**
+     * @param string $yuiBuildDirectory
+     * @param string $webDirectory
+     */
+    public function __construct($yuiBuildDirectory, $webDirectory)
+    {
+        $this->yuiBuildDirectory = $yuiBuildDirectory;
+        $this->webDirectory = $webDirectory;
+    }
+
+    public function getCombinedFileContent(array $files)
+    {
+        $this->assertAtLeastOneFile($files);
+
+        $content = '';
+
+        foreach ($files as $file) {
+            $ext = $this->findFileExtension($file);
+
+            if ($ext === 'js') {
```

Comment:
As a sequel to the comment above about getting the type first, then the content.

We could maybe simplify this a bit. Do the type consistency check in `getCombinedFilesContentType` only, and assume in this `getCombinedFileContent` that the list is consistent. We can update this method's description and say that it combines a valid list of files.

This would restrict type consistency checks in one place, and this method would become much simpler.